plugins/chat/app/services/chat/auto_join_channels.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+module Chat
+  class AutoJoinChannels
+    include Service::Base
+
+    ALLOWED_GROUP_PERMISSIONS ||= [
+      CategoryGroup.permission_types[:create_post],
+      CategoryGroup.permission_types[:full],
+    ]
+
+    policy :chat_enabled?
+
+    contract do
+      attribute :user_id, :integer
+      attribute :channel_id, :integer
+      attribute :category_id, :integer
+    end
+
+    step :create_memberships
+
+    private
+
+    def chat_enabled?
+      SiteSetting.chat_enabled
+    end
+
+    def create_memberships
+      automatic = ::Chat::UserChatChannelMembership.join_modes[:automatic]
+      group_permissions = ALLOWED_GROUP_PERMISSIONS
+      group_ids = SiteSetting.chat_allowed_groups_map
+      everyone_allowed = group_ids.include?(Group::AUTO_GROUPS[:everyone])
+      max_users = SiteSetting.max_chat_auto_joined_users
+      now = Time.zone.now
+      last_seen_at = 30.days.ago
+
+      # used to filter down to a specific user, chat channel, or category
+      user_sql = context.user_id.to_i > 0 ? "AND u.id = #{context.user_id}" : ""
+      channel_sql = context.channel_id.to_i > 0 ? "AND cc.id = #{context.channel_id}" : ""
+      category_sql = context.category_id.to_i > 0 ? "AND c.id = #{context.category_id}" : ""
```

Comment:
```suggestion
      user_sql = context.user_id > 0 ? "AND u.id = #{context.user_id}" : ""
      channel_sql = context.channel_id > 0 ? "AND cc.id = #{context.channel_id}" : ""
      category_sql = context.category_id > 0 ? "AND c.id = #{context.category_id}" : ""
```

We are already defining the attributes as integers in the contract so there is no need for `to_i` here.

Reply:
They're optional, so the `.to_i` is to convert `nil` values to `0`

Reply:
ooo icic. I thought a contract meant that the attribute has to be present. Thank you for clarifying 👍